components/google-cloud/google_cloud_pipeline_components/experimental/custom_job/custom_job.py

```diff
@@ -185,12 +190,62 @@ def _is_output_parameter(output_key: str) -> bool:
     if restart_job_on_worker_restart is not None:
         if 'scheduling' not in job_spec:
             job_spec['scheduling'] = {}
-        job_spec['scheduling']['restart_job_on_worker_restart'
-                              ] = restart_job_on_worker_restart
-    if service_account is not None:
-        job_spec['service_account'] = service_account
-    if network is not None:
-        job_spec['network'] = network
+        job_spec['scheduling'][
+            'restart_job_on_worker_restart'] = restart_job_on_worker_restart
+
+    # Remove any existing service_account from component input list.
+    input_specs[:] = [
+        input_spec for input_spec in input_specs
+        if input_spec.name != 'service_account'
+    ]
+    job_spec['service_account'] = "{{$.inputs.parameters['service_account}']}}"
+    input_specs.append(
+        structures.InputSpec(
+            name='service_account', type='String', default=service_account))
+
+    # Remove any existing network from component input list.
+    input_specs[:] = [
+        input_spec for input_spec in input_specs if input_spec.name != 'network'
+    ]
+    job_spec['network'] = "{{$.inputs.parameters['network}']}}"
+    input_specs.append(
+        structures.InputSpec(name='network', type='String', default=network))
+
+    # Remove any existing encryption_spec_key_name from component input list.
+    input_specs[:] = [
+        input_spec for input_spec in input_specs
+        if input_spec.name != 'encryption_spec_key_name'
+    ]
+    job_spec[
+        'encryption_spec_key_name'] = "{{$.inputs.parameters['encryption_spec_key_name}']}}"
+    input_specs.append(
+        structures.InputSpec(
+            name='encryption_spec_key_name',
+            type='String',
+            default=encryption_spec_key_name))
+
+    # Remove any existing tensorboard from component input list.
+    input_specs[:] = [
+        input_spec for input_spec in input_specs
+        if input_spec.name != 'tensorboard'
+    ]
+    job_spec['tensorboard'] = "{{$.inputs.parameters['tensorboard}']}}"
+    input_specs.append(
+        structures.InputSpec(
+            name='tensorboard', type='String', default=tensorboard))
```

Comment:
can we move all this code pattern to L259 similar to project/location? to be consistent

Reply:
done.